tensorflow/core/grappler/optimizers/constant_folding.cc

```diff
@@ -2460,9 +2468,33 @@ bool ConstantFolding::IsReductionSimplifiableToIdentity(
   return simplifiable;
 }
 
+bool ConstantFolding::ReplaceReductionWithIdentity(NodeDef* node) const {
+  // Replace the reduction node with an identity node, that can be further
+  // optimized by the model pruner.
+  DataType output_type;
+  if (node->attr().count("T") != 0) {
+    output_type = node->attr().at("T").type();
+  } else {
+    // This is an 'any' or 'all' reduction. The output is always boolean.
```

Comment:
Either check that it is indeed such a reduction or return false. Somebody might add new reduction types in the future. Alternatively state in the method-level docstring that this is a precondition that the caller should guarantee.

Reply:
Done. (I'll note that it's not my code; I only moved it :) )